libs/pika/async_cuda/src/cublas_exception.cpp

```diff
@@ -19,36 +19,49 @@ namespace pika::cuda::experimental {
             {
             case CUBLAS_STATUS_SUCCESS:
                 return "CUBLAS_STATUS_SUCCESS";
-            case CUBLAS_STATUS_NOT_INITIALIZED:
-                return "CUBLAS_STATUS_NOT_INITIALIZED";
-            case CUBLAS_STATUS_ALLOC_FAILED:
-                return "CUBLAS_STATUS_ALLOC_FAILED";
             case CUBLAS_STATUS_INVALID_VALUE:
                 return "CUBLAS_STATUS_INVALID_VALUE";
-            case CUBLAS_STATUS_ARCH_MISMATCH:
-                return "CUBLAS_STATUS_ARCH_MISMATCH";
-            case CUBLAS_STATUS_MAPPING_ERROR:
-                return "CUBLAS_STATUS_MAPPING_ERROR";
-            case CUBLAS_STATUS_EXECUTION_FAILED:
-                return "CUBLAS_STATUS_EXECUTION_FAILED";
             case CUBLAS_STATUS_INTERNAL_ERROR:
                 return "CUBLAS_STATUS_INTERNAL_ERROR";
-            case CUBLAS_STATUS_NOT_SUPPORTED:
-                return "CUBLAS_STATUS_NOT_SUPPORTED";
 #ifdef PIKA_HAVE_HIP
```

Comment:
Should we move the "common" error codes separately into the `PIKA_HAVE_HIP` and `PIKA_HAVE_CUDA` sections so that the returned strings are correct? I.e. so that we can have `rocblas_status_invalid_value` printed with HIP enabled instead of `CUBLAS_STATUS_INVALID_VALUE` etc.?

Reply:
Note, this comment does not invalidate my previous approval. Can be changed separately.

Reply:
yes definitely!

Reply:
I updated it in this PR